Google-Summer-of-Code-2025.md

```diff
@@ -1221,52 +1259,211 @@ This project entails introducing a developer-only UI (as part of the developer o
 </details>
 
-
 <details>
 <summary>Suggested PM demo points</summary>
 
 - Milestone 1: The new UI correctly shows all platform parameters and feature flags, and their correct value and sync statuses.
 
-- Milestone 2: The new screen fully supports overriding various platform parameters and feature flag values.
+- Milestone 2: The new screen fully supports overriding various platform parameters and feature flag values. This includes demonstrating both overriding user flows in the new UI, and showing the value changes correctly take effect in the app after a prompted (and accepted) app restart.
+</details>
+
+
+
+### 4.3. Android lint infrastructure & fixes
+
+**Project Description:**
+
+The Android build ecosystem supports a [linting tool](https://developer.android.com/studio/write/lint) which catches a variety of Android-specific problems (beyond those that can be caught by more general-purpose linters like [ktlint](https://github.com/pinterest/ktlint)). Historically a Gradle feature, Android Lint is [not yet](https://github.com/bazelbuild/rules_android/issues/321) directly supported in the Bazel world (though there are some [alternatives available](https://github.com/Bencodes/rules_android_lint)). This project involves:
+- Introducing an Oppia Android script that runs Android lint directly.
+- Supporting the same exemption override support (via XML) but using a textproto interface for parity with the allow-listing and deny-listing configurations used in other Oppia Android scripts.
+- Pre-populating the exemption list with existing known failures.
+- Adding wiki documentation to explain how to use the new script.
+- Filing and/or updating issues with findings such that all remaining Android lint issues are properly tracked in the repository.
+- Updating CI to run the Android lint script.
+- Fixing the following categories of lint issues:
+  - All categories classified as 'errors' by the tool.
+  - All of the following 'warning' categories:
+    - OldTargetApi
+    - UnusedAttribute
+    - Typos
+    - StringFormatCount
+    - ObsoleteSdkInt
+    - UnusedResources
+    - UselessLeaf
+    - UselessParent
+    - UnusedIds
+    - DuplicateStrings
+    - SelectableText
+    - LabelFor
+    - RtlSymmetry
+    - UnknownNullness
+
+**Tracking issues**: [#5734](https://github.com/oppia/oppia-android/issues/5734)
+
+**Size:** Large (\~350 hours)
+
+**Difficulty**: Hard
+
+**Potential mentors:** @BenHenning
+
+**Product/technical clarifiers:** @BenHenning (product + technical)
+
+**Required knowledge/skills:**
+- Figure out the root cause of an issue and communicate it well using a [debugging doc](https://github.com/oppia/oppia/wiki/Debugging-Docs).
+- Build the app and install it on a local device or emulator. Then verify that you can (a) build a non-test library  target locally, (b) run a unit test locally, and (c) play through the app locally.
+- Write new Kotlin code with unit tests.
+
+**Related issues:**
```

Comment:
https://github.com/oppia/oppia-android/issues/5169 tracks known lint categories

Reply:
Good call--added and also noted that the categories in that issue aren't actually all of the ones that will be handled in the PR (or supported by the script).

Reply:
This comment is fine to resolve.